contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -742,51 +768,152 @@ have a built-in `nil` value.
+external state (such as annotations on StorageClass objects) when deciding
+default values.  These admission controllers are strictly optional and can be
+disabled.  As such, fields which are initialized this way must be strictly
+optional.
 
-## Late Initialization
+Like static defaults, these are run synchronously to the API operation in
+question, and when the API call completes, all static defaults will have been
+set.  Subsequent GETs of the resource will include the default values
+explicitly.
+
+### Controller-Assigned Defaults (aka Late Initialization)
 
 Late initialization is when resource fields are set by a system controller
-after an object is created/updated.
+after an object is created/updated (asynchronously).  For example, the
+scheduler sets the `pod.spec.nodeName` field after the pod is created.
+
+Like admission controlled defaults, these controllers may consider external
+state when deciding default values and can be disabled.  Fields which are
+initialized this way must be strictly optional.
+
+Like all controllers, care must be taken to not clobber unrelated fields or
+values (e.g. in an array).  Using one of the patch or apply mechanisms is
+recommended to facilitate composition and concurrency of controllers.
 
-For example, the scheduler sets the `pod.spec.nodeName` field after the pod is
-created.
+### What May Be Defaulted
 
-Late-initializers should only make the following types of modifications:
+All forms of defaulting should only make the following types of modifications:
  - Setting previously unset fields
  - Adding keys to maps
  - Adding values to arrays which have mergeable semantics
-(`patchStrategy:"merge"` attribute in the type definition).
-
-These conventions:
- 1. allow a user (with sufficient privilege) to override any system-default
- behaviors by setting the fields that would otherwise have been defaulted.
- 1. enables updates from users to be merged with changes made during late
-initialization, using strategic merge patch, as opposed to clobbering the
-change.
- 1. allow the component which does the late-initialization to use strategic
-merge patch, which facilitates composition and concurrency of such components.
-
-Although the apiserver Admission Control stage acts prior to object creation,
-Admission Control plugins should follow the Late Initialization conventions
-too, to allow their implementation to be later moved to a 'controller', or to
-client libraries.
+   (`patchStrategy:"merge"` attribute in the type definition).
+
+In particular we never want to change or override a value that was provided by
+the user.  If they requested something invalid, they should get an error.
+
+This:
+ 1. allows a user (with sufficient privilege) to override any system-default
+    behaviors by explicitly setting the fields that would otherwise have been
+    defaulted.
+ 1. enables updates from users to be merged with default values.
+
+### Considerations For PUT Operations
```

Comment:
I like this discussion you had in this issue about PUT https://github.com/kubernetes/kubernetes/issues/91459#issuecomment-780436664, I think that is the source of this section, right?

maybe you can reuse some of the comments and examples there, 

>  The general rule about PUT is that PUT on an existing resource should adjust that resource to be in the same working configuration as if you used that PUT to create a new resource. Calling PUT a second time with the same input should be acceptable and should not change the resource.

Reply:
I will update the text here.